### PR TITLE
Fetch user bikes

### PR DESCRIPTION
**Pull request by timlkelly**
- Add job that retrieves a user's bikes from bike index upon log in
- Add job to check if there are any valid serial_searches for that bike
- Add rake task to perform these jobs

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -16,6 +16,7 @@
 # users commonly want.
 #
 # See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
+require 'capybara/rspec'
```
Comment (sethherr) [inline spec/spec_helper.rb]: wait wat?

Why did you add capybara?

File: spec/cassettes/fetch_bike_index_bikes_job_user_log_in.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      - max-age=31536000
+      X-Powered-By:
+      - cloud66
+      Server:
+      - cloudflare-nginx
+      Cf-Ray:
+      - 337eaa496c9710b7-ORD
+    body:
+      encoding: ASCII-8BIT
+      string: '{"access_token":"bbafcc8a28835fedb99cee9f67f941502dd6c72ad2a6548fcbabf540dac30198","token_type":"bearer","expires_in":3600,"refresh_token":"f3f94cb45fa378883b686693f8742eec84866eb25177976df3cd518826cf0617","scope":"read_bikes
```
Comment (sethherr) [inline spec/cassettes/fetch_bike_index_bikes_job_user_log_in.yml]: Is this your access token? Because that's bad.

File: spec/fixtures/omniauth_binx_response.json [hunk 3]
```diff
@@ -13,9 +13,9 @@
     "image": "/uploads/Us/1/bike.png"
   },
   "credentials": {
-    "token": "9f157d7bb3cf73846b0def9700f2e16800b0ae228cf52ead778f3f4ee44358ea",
-    "refresh_token": "57b0077de9b0db5b4b5761d7e177436a3c26df6a5e5374f1dde5b290f49f7e93",
-    "expires_at": 1429800712,
+    "token": "18e7e5427889c087cd7dcdc5282f2128c35bf7b5771f9cfae2ea3d65106dd555",
```
Comment (sethherr) [inline spec/fixtures/omniauth_binx_response.json]: Is this a random string? Or is it an actual access token?

File: Gemfile [hunk 4]
```diff
@@ -63,13 +63,14 @@ end
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platform: :mri
-  gem 'rspec-rails'
-  gem 'shoulda-matchers'
-  gem 'guard-rspec'
-  gem 'guard-rubocop'
+  gem 'capybara'
+  gem 'dotenv-rails'
```
Comment (sethherr) [inline Gemfile]: Try to avoid moving things around unnecessarily. I can't tell what you added easily here. Is it just capybara?

File: spec/fixtures/omniauth_binx_response.json (hunk 3 above)
Comment (sethherr) [inline spec/fixtures/omniauth_binx_response.json]: And that `refresh_token` is particularly bad

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 5]
```diff
@@ -5,6 +5,7 @@ def bike_index
     @user = User.from_omniauth(request.env['omniauth.auth'].uid, request.env['omniauth.auth'])
     @user.update_binx_credentials(request.env['omniauth.auth'])
     if @user.persisted?
+      FetchBikeIndexBikesJob.perform_later(@user)
```
Comment (sethherr) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Way better place to put this than in the authentication block!

File: spec/jobs/check_bike_serial_searches_job_spec.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe CheckBikeSerialSearchesJob, type: :job do
+  describe 'perform' do
+    let(:serial_search) { FactoryGirl.create(:serial_search) }
+    let!(:bike) { FactoryGirl.create(:bike, serial: serial_search.serial) }
```
Comment (sethherr) [inline spec/jobs/check_bike_serial_searches_job_spec.rb]: Nice use of the bang let method here!

**Comment by timlkelly** [comment]
Yup seems to be working locally.

**Comment by sethherr** [comment]
👍
